### ISSUE TITLE
About training-testing strategies of downstream datasets

### ISSUE THREAD

**Issue by jsadu826** [Luffy03/VoCo#19]
Hello! I'm trying to reproduce the result on CC-CCII and MM-WHS.

-  For CC-CCII, what is the training-validation-testing setting? Is it a 3-fold cross-validation?
-  For MM-WHS, is the testing done by running the matlab script (matlab_evaluation_script.m) provided by the authors on the encrypted testing labels?

Thank you!

**Comment by Luffy03** [comment]
Hi, thanks for your attention to our work!
1. For CC-CCII, you can three-fold validation and report the mean results. In my experiments, the results of three folds are almost the approximate. You can have a try.
2. For WHS, since most of the existing works did not report their test results. For convenience, you can also report 5-fold validation results on the training set.

**Comment by jsadu826** [comment]
For CC-CCII data pre-processing, for each scan of each patient, are all the PNGs stacked to get a 3D npy volume with shape (n_slices, n_channels, height, width)? Do weed need to select only the lesion slices? When training, are the volumes resized to (n_slices, n_channels, 256, 256) and directly sent to the model, without further cropping into fixed-size ROIs (e.g. 64x64x64) as in the segmentation tasks?

**Comment by Luffy03** [comment]
We don't need to select the lesion slice. Instead, we input 3D volume to train a 3D network. Please refer to  "https://github.com/Luffy03/VoCo/blob/main/Finetune/CC-CCII/utils/data_utils.py" for details.

**Comment by jsadu826** [comment]
How is the BraTS 2021 dataset preprocessed in training and testing and which modality (t1, t1ce, t2, flair) is used.

**Comment by Luffy03** [comment]
Hi, we used all of four modalities.

**Comment by jsadu826** [comment]
So the model input has 4 channels?

**Comment by Luffy03** [comment]
Yes, exactly. And we don't load the first layer of our pre-trained models.

**Comment by jsadu826** [comment]
Great thanks!

**Comment by jsadu826** [comment]
Also, could you please share the train-valid-test json for BraTS21? Thank you~

**Comment by Luffy03** [comment]
Hi, the json file of BraTS21 is copied from https://drive.google.com/file/d/1i-BXYe-wZ8R9Vp3GXoajGyqaJ65Jybg1/view?usp=sharing. We will also provide our implementation of BraTS21 recently.

**Comment by jsadu826** [comment]
So VoCo used 5-fold crossvalidation for BraTS21, as in https://arxiv.org/pdf/2201.01266？

**Comment by Luffy03** [comment]
Yes, you can have a try. In my experiments, the results of different folds are approximate, since BraTS21 is already with adequate cases . I notice that some previous works report the results of the first fold, maybe 5-fold is not an essential setting.

**Comment by jsadu826** [comment]
> Yes, exactly. And we don't load the first layer of our pre-trained models.

Does it mean not loading **swinViT.patch_embed.proj.weight**, **encoder1.layer.conv1.conv.weight**, and **encoder1.layer.conv3.conv.weight**?

**Comment by Luffy03** [comment]
Hi, you can use this code to check.
```
def load(model, model_dict):
    if "state_dict" in model_dict.keys():
        state_dict = model_dict["state_dict"]
    elif "network_weights" in model_dict.keys():
        state_dict = model_dict["network_weights"]
    elif "net" in model_dict.keys():
        state_dict = model_dict["net"]
    else:
        state_dict = model_dict

    if "module." in list(state_dict.keys())[0]:
        print("Tag 'module.' found in state dict - fixing!")
        for key in list(state_dict.keys()):
            state_dict[key.replace("module.", "")] = state_dict.pop(key)

    if "backbone." in list(state_dict.keys())[0]:
        print("Tag 'backbone.' found in state dict - fixing!")
    for key in list(state_dict.keys()):
        state_dict[key.replace("backbone.", "")] = state_dict.pop(key)

    if "swin_vit" in list(state_dict.keys())[0]:
        print("Tag 'swin_vit' found in state dict - fixing!")
        for key in list(state_dict.keys()):
            state_dict[key.replace("swin_vit", "swinViT")] = state_dict.pop(key)

    current_model_dict = model.state_dict()
    new_state_dict = {
        k: state_dict[k] if (k in state_dict.keys()) and (state_dict[k].size() == current_model_dict[k].size()) else current_model_dict[k]
        for k in current_model_dict.keys()}

    model.load_state_dict(new_state_dict, strict=True)
    print("Using VoCo pretrained backbone weights !!!!!!!")

    return model
```

**Comment by jsadu826** [comment]
> Hi, the json file of BraTS21 is copied from https://drive.google.com/file/d/1i-BXYe-wZ8R9Vp3GXoajGyqaJ65Jybg1/view?usp=sharing. We will also provide our implementation of BraTS21 recently.

Hi, I'd like to know why the BraTS21 results obtained by training SwinUNETR from scratch are much higher in [SwinUNETR](https://arxiv.org/pdf/2201.01266) than in VoCo, especially the enhanced tumor. These two works seems to have used the same data split.

In [SwinUNETR](https://arxiv.org/pdf/2201.01266) paper:
![image](https://github.com/user-attachments/assets/dd38256f-491a-4a4c-a327-b7c00d2a52ef)

In VoCo paper:
![image](https://github.com/user-attachments/assets/156b5eec-cfeb-403c-ab0a-b9ae2a167e03)

**Comment by Luffy03** [comment]
Hi, it is a mistake in our previous implementation :worried: since we inherited the implementation of swinunetr's official codes in the cvpr version. Our current version can achieve higher performance and we will release it recently. By the way, can you reproduce the results as the swinunetr reported? We cannot reproduce it.

**Comment by jsadu826** [comment]
I copied the data split from [this IEEE TMI paper](https://arxiv.org/pdf/2207.07370), which splits the 1251 BraTS21 training data into train/valid/test = 833/209/209. The training code is based on [](https://github.com/Project-MONAI/research-contributions/tree/main/SwinUNETR/BRATS21). The dices on the testing set were approximately:  TC = 90, WT = 93, ET = 86 (almost same for both finetuning on VoCo and training from scratch, but finetuning on VoCo indeed converged much much faster).
![image](https://github.com/user-attachments/assets/ce063b5b-8e77-428e-adac-db0a4b10a6a1)

**Comment by Luffy03** [comment]
Thanks for sharing, encouraging to hear that. Our reproduced result is about 91% Dice but it is not based on [this](https://github.com/Project-MONAI/research-contributions/tree/main/SwinUNETR/BRATS21). Seems your results are also good. We also find that VoCo did not improve significantly in this dataset (less than 2%). Maybe it is caused by the modality gap.

**Comment by Luffy03** [comment]
Dear researchers, our work is now available at [**Large-Scale-Medical**](https://github.com/Luffy03/Large-Scale-Medical), if you are still interested in this topic. Thank you very much for your attention to our work, it does encourage me a lot!
